#!/usr/bin/python
PACKAGE = "dynamic_mapping_params"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("l_elbow_scale",       double_t, 0, "A double parameter",   1.0,  0.5, 2.0)
gen.add("l_elbow_offs_x",      double_t, 0, "A double parameter",   0.0, -0.7, 0.7)
gen.add("l_elbow_offs_y",      double_t, 0, "A double parameter",   0.2, -0.7, 0.7)
gen.add("l_elbow_offs_z",      double_t, 0, "A double parameter",   0.2, -0.7, 0.7)

gen.add("r_elbow_scale",       double_t, 0, "A double parameter",   1.0,  0.5, 2.0)
gen.add("r_elbow_offs_x",      double_t, 0, "A double parameter",   0.0, -0.7, 0.7)
gen.add("r_elbow_offs_y",      double_t, 0, "A double parameter",  -0.2, -0.7, 0.7)
gen.add("r_elbow_offs_z",      double_t, 0, "A double parameter",   0.2, -0.7, 0.7)

gen.add("l_wrist_scale",       double_t, 0, "A double parameter",   1.0,  0.5, 2.0)
gen.add("l_wrist_offs_x",      double_t, 0, "A double parameter",   0.0, -0.7, 0.7)
gen.add("l_wrist_offs_y",      double_t, 0, "A double parameter",   0.2, -0.7, 0.7)
gen.add("l_wrist_offs_z",      double_t, 0, "A double parameter",   0.2, -0.7, 0.7)

gen.add("r_wrist_scale",       double_t, 0, "A double parameter",   1.0,  0.5, 2.0)
gen.add("r_wrist_offs_x",      double_t, 0, "A double parameter",   0.0, -0.7, 0.7)
gen.add("r_wrist_offs_y",      double_t, 0, "A double parameter",  -0.2, -0.7, 0.7)
gen.add("r_wrist_offs_z",      double_t, 0, "A double parameter",   0.2, -0.7, 0.7)

exit(gen.generate(PACKAGE, "dynamic_mapping_params", "MappingParams"))